titus-ext/aws/src/main/java/com/netflix/titus/ext/aws/AmazonClientProvider.java

```diff
@@ -0,0 +1,65 @@
+package com.netflix.titus.ext.aws;
+
+import java.util.concurrent.ConcurrentHashMap;
+import java.util.concurrent.ConcurrentMap;
+import javax.inject.Inject;
+import javax.inject.Singleton;
+
+import com.amazonaws.arn.Arn;
+import com.amazonaws.auth.AWSCredentialsProvider;
+import com.amazonaws.auth.STSAssumeRoleSessionCredentialsProvider;
+import com.amazonaws.services.elasticloadbalancingv2.AmazonElasticLoadBalancingAsync;
+import com.amazonaws.services.elasticloadbalancingv2.AmazonElasticLoadBalancingAsyncClientBuilder;
+import com.amazonaws.services.securitytoken.AWSSecurityTokenServiceAsync;
+
+@Singleton
+public class AmazonClientProvider {
+
+    private final AwsConfiguration configuration;
+    private final AWSSecurityTokenServiceAsync stsClient;
+
+    private ConcurrentMap<String, AWSCredentialsProvider> awsCredentialsByAccountId = new ConcurrentHashMap<>();
+    private ConcurrentMap<String, AmazonElasticLoadBalancingAsync> loadBalancerClients = new ConcurrentHashMap<>();
+
+    @Inject
+    public AmazonClientProvider(AwsConfiguration configuration,
+                                AWSSecurityTokenServiceAsync stsClient) {
+        this.configuration = configuration;
+        this.stsClient = stsClient;
+    }
+
+    public AmazonElasticLoadBalancingAsync getLoadBalancingClient(String accountId) {
+        return loadBalancerClients.computeIfAbsent(accountId, id -> {
```

Comment:
How do we make sure the credentials provider in the Map is still valid and has not expired the session duration when first constructed?

Reply:
The `STSAssumeRoleSessionCredentialsProvider` has a background thread that does the automatic refreshing.